packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -267,24 +410,22 @@ export class BlobManager {
                 this.redirectTable.delete(blobId);
                 continue;
             }
-            this.blobIds.delete(blobId);
         }
     }
 
     public summarize(): ISummaryTreeWithStats {
-        // If we have a redirect table it means the container is about to transition to "Attaching" state, so we need
-        // to return an actual snapshot containing all the real storage IDs we know about.
-        const attachingOrAttached = !!this.redirectTable || this.runtime.attachState !== AttachState.Detached;
-        const blobIds = attachingOrAttached ? this.blobIds : this.detachedBlobIds;
+        // if storageIds is empty, it means we are detached and have only local IDs, or that there are no blobs attached
+        const blobIds = this.storageIds.size > 0 ? Array.from(this.storageIds) : Array.from(this.redirectTable.keys());
```

Comment:
It looks like we add all storageIds to this.redirectTable as storageId:storageId pairs. So, we can simply always just use `this.redirectTable.keys()` which will include both local and storageIds.

Reply:
In the current implementation, any lookup for a storageId has to be done in both redirect table and storageIds. For example, when GC is deleting an id, it has to do in both these places. If we have a single source of truth and use that exclusively, the look up only has to be done in one place.